dot-studio/studio_setup

```diff
@@ -38,3 +38,60 @@ function generate_netrc_config() {
     echo -e "machine github.com\n  login $GITHUB_TOKEN" >> ~/.netrc
   fi
 }
+
+
+document "clone_function" <<DOC
+  Create a fork of the ci-studio-common function in your .studio/ directory.
+
+  If there is a function, provided by ci-studio-common, that isn't working exactly
+  as required, you can use this function to create a copy of that function in your
+  local .studio/ directory. You can then modify or manipulate that function declaration
+  as necessary to meet your needs.
+
+  From that point forward, anytime you open your studio, your copy of that function will
+  be used, rather than the version that comes with ci-studio-common.
+
+  If you need to use it immediately, you can source that file.
+
+  . /src/.studio/override-common
+
+  We encourage you to, if appropriate, contribute your changes back to ci-studio-common.
+DOC
+function clone_function() {
+  local function="$1"
+
+  # Return an error if that is not a defined function
+  if [[ ! -f /tmp/docs/$function ]]
+  then
+    echo "ERROR: ci-studio-common does not provide a '$function' function."
+  fi
+
+  # Create the .studio directory if it's missing
+  [[ ! -d /src/.studio ]] && mkdir /src/.studio
```

Comment:
[nit] Instead of checking for the dir it is easier just to call `mkdir -p ...` since the command will create it if only if it doesn't exist.